plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/proposals/imports/ImportsAwareReferenceProposalCreator.java

```diff
@@ -102,6 +103,10 @@ public void lookupCrossReference(
 			Function<IEObjectDescription, ICompletionProposal> proposalFactory) {
 
 		if (model != null) {
+			if (PickyCompletionProposalAcceptor.hasSeenScopeFor(acceptor, model, reference)) {
```

Comment:
Here you could try to call `org.eclipse.xtext.ui.editor.contentassist.AbstractJavaBasedContentProposalProvider.announceProcessing(List<?>)` with list containing a unique first element, e.g. `"lookupCrossReference"` and the model and reference. That should already do the trick for a given ContentAssistContext.